appsec-kit-starter/src/main/java/com/vaadin/appsec/service/AbstractInitListener.java

```diff
@@ -0,0 +1,31 @@
+/*-
+ * Copyright (C) 2023 Vaadin Ltd
+ *
+ * This program is available under Vaadin Commercial License and Service Terms.
+ *
+ *
+ * See <https://vaadin.com/commercial-license-and-service-terms> for the full license.
+ */
+package com.vaadin.appsec.service;
+
+import com.vaadin.flow.server.VaadinService;
+import com.vaadin.flow.server.VaadinServiceInitListener;
+
+/**
+ * Base class for all classes implementing {@link VaadinServiceInitListener} in
+ * the AppSec Kit.
+ */
+public abstract class AbstractInitListener
+        implements VaadinServiceInitListener {
```

Comment:
Instead of having an abstract superclass + two different listener, why not having a single listener for both initialization and notifications? It's not too much code to justify splitting in subclasses and also helps to ensure we first initialize the service before notifications.

Reply:
Good suggestion. I combined them into one class using only one listener.